### DISCUSSION TITLE
Request for Homomorphic Comparison of Two Cipher Texts Sample Code

### DISCUSSION THREAD

**Discussion by PavanNPatil17** [tuneinsight/lattigo#354] [Ideas]
Hello everyone,

I am currently working on a project that requires me to perform homomorphic comparison of two cipher texts. I have been searching for sample code that can help me understand the implementation of this process, but I haven't found anything useful yet.

I would like to request anyone who has worked on a similar project or has knowledge in this field to kindly share some sample code or resources that could help me with this task. I am particularly interested in code examples using homomorphic encryption schemes or fully homomorphic encryption (FHE).

Any help would be greatly appreciated, and I believe that it will not only help me but also benefit other researchers who are working on similar projects.

Thank you in advance for your support.

Best regards,
Pavan N Patil

**Comment by Pro7ech** [comment]
Hi @PavanNPatil17,

High accuracy homomorphic comparison is a planned feature for Lattigo. This will include the work of  [Lee et al.](https://eprint.iacr.org/2020/834). However there is still quite some work to finish before we can integrate this into the library. 

In the mean time, I can provide you with a [sample of composite polynomials](https://pastebin.com/Fbxhgztv) computed using the work of Lee et al. They allow to approximate the sign function in the interval `[-1, 2^-a] U [2^-a, 1]` for `1 <= a <= 30`.

Using the sign function you can implement a multiplexer gate `MUX(a, b, sign(a, b)) -> (min(a, b), max(a, b))` that will sort two values. This can be used as a building block for larger sorting (e.g. as in the work of [Hong et al.](https://eprint.iacr.org/2021/551)).

Beware that such operation is expensive and will require multiple bootstrappings per comparison.

Hope this help :)
